geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/MemberIdentifierImpl.java

```diff
@@ -676,7 +675,14 @@ public void toDataPre_GFE_9_0_0_0(DataOutput out, SerializationContext context)
         durableId == null ? 300 : memberData.getDurableTimeout(),
         out);
 
-    short version = memberData.getVersionOrdinal();
+    final short version;
+    if (context.getSerializationVersion().isNewerThan(KnownVersion.GFE_90)) {
+      version = memberData.getVersionOrdinal();
+    } else {
+      // This version of geode-native could not parse the longer version ordinals.
+      // Let's make these older clients think the server is still 1.14.
+      version = KnownVersion.GEODE_1_14_0.ordinal();
```

Comment:
This probably should use the actual member version, or 1.14.0, whichever is older. As it is, this is going to make some ids (maybe from a client, or a peer, or a WAN site) look NEWER than they actually are.

Reply:
This code will only be in 1.15+, so 1.14.0 will always be the older.

Reply:
The version here is the version of this MemberIdentifierImpl object, which may or may not be the member currently executing this code. It could be a MemberIdentifierImpl from a peer, or even from a client or WAN site.

Reply:
I see what you are saying now. Yeah, I can make that change and see if it affects tests.